packages/experimental/src/inbox-note/action.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { ADMIN_URL as adminUrl } from '@woocommerce/wc-admin-settings';
+
+type InboxNoteActionProps = {
+	onClick: () => void;
+	label: string;
+	href?: string;
+};
+
+/**
+ * Renders a secondary button that can also be a link. If href is provided it will
+ * automatically open it in a new tab/window.
+ */
+const InboxNoteAction: React.FC< InboxNoteActionProps > = ( {
+	label,
+	onClick,
+	href,
+} ) => {
+	const [ inAction, setInAction ] = useState( false );
+
+	const handleActionClick: React.MouseEventHandler = ( event ) => {
```

Comment:
If you use the generic arg here you can combine it with `currentTarget` and remove the type narrowing. 

```suggestion
	const handleActionClick: React.MouseEventHandler<HTMLAnchorElement> = ( event ) => {
```